test/persistence/compaction/compaction.td

```diff
@@ -21,147 +21,27 @@ $ set-sql-timeout duration=30s
     WHERE metric = grp.metric
     ORDER BY time DESC LIMIT 1)
 
-# At the beginning there are no trace batches.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-0
-
 > CREATE TABLE compaction (f1 INTEGER);
 
-> INSERT INTO compaction VALUES (1);
-
-# The insert should make it into trace, and form 1 trace batch.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-1
-
-# Read after write to coerce different writes to happen at different timestamps.
-> SELECT * FROM compaction
-f1
-----
-1
-
-> INSERT INTO compaction VALUES (2);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-
-# The previous insert should get compacted into one batch with the one before it.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-1
-
-> INSERT INTO compaction VALUES (3);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-3
-
-# The most recent insert forms a new batch because it is at a lower compaction level.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-2
-
-> INSERT INTO compaction VALUES (4);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-3
-4
-
-# The previous insert should compact together with the previous batch, which then compacts recursively
-# up, leaving just one batch.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
+# At the beginning there's fewer than 200 bytes in the stored arrangement.
```

Comment:
it's probably worth sticking a comment in the file somewhere that this test is only attempting to check that persist compaction is correctly hooked up to user tables. all other compaction tests live in src/persist

(this reminds me, it might be worth our while to have a variant of this test that does the same thing but through kafka to check that sources are hooked up to compaction)

Reply:
added the comment, going to punt on the kafka test for a followup pr